File: Day30/frontend/src/domain/entities/UserProfile.ts
// matching what the backend's UserProfileDto actually returns.
export interface UserProfile {
  userId: string;
  email: string;
  fullName: string;
  address: string;
  gender?: string | null;
  phoneNumbers: string[];
  avatarUrl?: string | null;
  memberNumber: number;
  isActive: boolean;
}

File: Day30/Form(backend)/Entities/UserProfile.cs
namespace Form.Entities;

public class UserProfile
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public Guid UserId { get; set; }
    public User User { get; set; } = null!;

    public string FullName { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public Gender? Gender { get; set; }
    public List<string> PhoneNumbers { get; set; } = new();
    public string? AvatarUrl { get; set; }

    // No setter logic needed here — the database assigns this automatically
    // on insert, starting at 1 and counting up. We never set this in C# code.
    public int MemberNumber { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}